src/middlewares/problem.middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { logger } from '../utils/logger';
+
+/**
+ * Catch problem exception, log it and serialize error to human readable form.
+ */
+export function problemMiddleware(error: ProblemException, req: Request, res: Response, next: NextFunction) {
+  try {
+    const status = error.status = error.status || 500;
+    error.title = error.title || 'Internal server error';
+
+    logger.error(`[${req.method}] ${req.path} >> Status:: ${status}, Type:: ${error.type?.replace('https://api.asyncapi.com/problem/', '')}, Title:: ${error.title}, Detail:: ${error.detail}`);
+
+    const problem = ProblemException.toJSON(error, status >= 500);
+    res.status(status).json(problem);
```

Comment:
As it's the first time I see ExpressJS code, I need to check documentation to understand few things. I have a doubt here:

ExpressJS docs say:
> So when you add a custom error handler, you must delegate to the default Express error handler, when the headers have already been sent to the client:

Are we doing this (calling the default error handler)?

Reply:
Hood point! I will add this example solution from docs 👍🏼